incubator/hnc/internal/validators/object.go

```diff
@@ -104,6 +104,14 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, inst *unstructured
 	return allow("")
 }
 
+func (o *Object) isPropagateType(inst *unstructured.Unstructured) bool {
+	o.Forest.Lock()
+	defer o.Forest.Unlock()
+
+	ts := o.Forest.GetTypeSyncer(inst.GroupVersionKind())
```

Comment:
Hmm. This means that if we're propagating CRD v1, and someone tries to make a change to v1beta1, we won't run the webhook since we won't have a reconciler for v1beta1. So I think we need to change this lookup to be based on GroupKind, not GroupVersionKind. sgty?

Reply:
Hmm, if we don't have a reconciler for v1beta1, then objects of v1beta1 shouldn't be propagated, right? Since we only deny changing the labels in any type that is in propagate mode. Even if for example, v1.Secret is removed from the type configuration and the previous propagated objects are left and we allow changes to those obsolete propagated objects, right?

Reply:
No, that's not how it works. v1beta1 and v1 are two different views _onto the same object_. So when we say "propagate example.com/Foobar v1", what that actually means is "please use a v1 client to propagate objects of type Foobar." But _other_ clients could be reading and writing _the same objects_ usina a v1beta1 or even v2 API.

So imagine that there's a "Secret v2" that's added to K8s, which is a superset of the features available in Secret v1. The HNC admin would have to change the config to "Secret v2" to make sure the new fields get copied - they can't have _both_ v1 and v2 at the same time. But that means that if another client tries to modify a secret using the v1 interface, the webhook won't be applied (we'll early exit).

Does this make sense?

Reply:
Ah right got you, yes we just discussed. Thanks, will update it too!